openzwavemqtt/const.py

```diff
@@ -237,6 +237,8 @@ class ValueIndex(IntEnum):
     # Notification Event Parameter
     NOTIFICATION_EVENT_PARAMETER_PREVIOUS_EVENT_CLEARED = 256
     NOTIFICATION_EVENT_PARAMETER_USER_CODE = 260
+    USERCODE_120 = 120
+    USERCODE_121 = 121
```

Comment:
Reviewing the manual, these 120/121 are configuration parameters.
Why would you need these defined as a ValueIndex?

Reply:
To be able to set them by using set_config_parameter

Reply:
That has to do with the XMLs in OpenZWave not the `ValueIndex` in this library.

Reply:
I can set the values directly in OZWAdmin just fine. 
Without this the node.get_value(CommandClass.CONFIGURATION, 120) returns None. Adding them here makes set_config_parameter work for them.

Reply:
I tried this just to make sure:
![image](https://user-images.githubusercontent.com/974569/96051053-9d1e4780-0e72-11eb-918e-220aab2cccc1.png)

This is what it prints with this PR:
![image](https://user-images.githubusercontent.com/974569/96051114-be7f3380-0e72-11eb-8d44-5b0ca989ff2c.png)

And without:
![image](https://user-images.githubusercontent.com/974569/96051145-d35bc700-0e72-11eb-8a65-0fd6da91338a.png)

Reply:
What does the node's XML entry look like in the ozwcache xml?

Reply:
My question is what happens here:

https://github.com/cgarwood/python-openzwave-mqtt/blob/25f9f90ba586e33fee7b7a444743966371268787/openzwavemqtt/models/command_class.py#L57-L63

Is `value` a `OZWValue`? If yes, `OZWValue.index` returns `ValueIndex.UNKNOWN` if it can't cast the mqtt `"Index"` into a `ValueIndex`. 

https://github.com/cgarwood/python-openzwave-mqtt/blob/25f9f90ba586e33fee7b7a444743966371268787/openzwavemqtt/models/value.py#L66-L76

The values 120, 121, etc. are not defined in `ValueIndex`, so it probably returns `ValueIndex.UNKNOWN` and then `None` since it never matches the integer `value_index` argument. There are a lot of numbers defined though, so it probably works for other many other parameters.

This PR is probably not the right fix though. It happens to work just because it creates matching values in `ValueIndex`, but is not a general solution.

If I got that right, this would also explain [this problem](https://community.home-assistant.io/t/openzwave-administration/229436/10?u=freshcoast) posted in the community.

Reply:
Also `ValueIndex` doesn't really make much sense for Configuration CC because none of the indices are assigned according to some specification, whether it's Z-Wave or OpenZWave. So while `ValueIndex.SENSOR_MULTILEVEL_AIR_TEMPERATURE = 1` is a fixed index value defined by OpenZWave/Z-Wave (for the Sensor Multilevel CC at least), a Configuration CC value with index = 1 can mean something different for each device out there.

Reply:
> If I got that right, this would also explain [this problem](https://community.home-assistant.io/t/openzwave-administration/229436/10) posted in the community.
Yes, that's the error I get without the values added in ValueIndex. 
If config values should not go there, then the set_config_parameter method is not done correctly, since as you mentioned it uses get_value, which returns None if the index is not on ValueIndex. If config values should not be in ValueIndex, then why is set_config_parameter using ValueIndex?

Reply:
We're open to suggestions to improve the design around this.

Reply:
Right now set_config_parameter depends on the index being on ValueIndex, so until a better solution comes up, config indexes should be added there IMHO.

Reply:
I don't think we need to rush this. We should think about how we want the design to be and then make that.

Reply:
Perhaps if the CC = `Configuration CC` ignore the `valueindex` and return the `self.data.get("Index")`?

Reply:
I don't know how the typing works, `OZWValue.index` wants to return a `ValueIndex`. Can it be typed to return either an `int` or `ValueIndex`? If so, it could just return the int for an unknown value. A `ValueIndex` is comparable to an `int`, right?

You could also change `get_value_by_index` to skip using the `ValueIndex` or add a property to get the integer value. `get_value_by_index` types the index as `ValueIndex`, but its callers allow `int`.

```python
 def get_value_by_index(self, value_index: ValueIndex) -> Optional[OZWValue]: 
     """Return a specific OZWValue on this CommandClass (if exists).""" 
     # pylint: disable=no-member 
     for value in self.values(): 
#         if value.data.get("Index") == value_index: 
         if value.index_raw == value_index: 
             return value 
     return None
```

The code I referenced in my comment above has a comment about having ValueIndex values defined for each CC, how would that be done?
```
# TODO: we can make this prettier by returning the 
# ValueIndex belonging to this specific CC 
# Now it will return the first value matching this int value 
# Nothing breaks but it isn't very pretty 
```

FWIW, you could say OpenZWave takes the similar approach to this PR, that is for most (?) value types it defines all possible values. It is a different enum per value though, not a single enum that covers all possible values. [Here](https://github.com/OpenZWave/open-zwave/blob/b0afd4c60b38065fdbc57835cb3eb6035afcc00f/cpp/src/ValueIDIndexesDefines.def#L1893) is the enum for config parameters, for example:
```c++
ENUM(ValueID_Index_Configuration,
		Parameter_1 = 1,
		Parameter_2,
		Parameter_3,
		Parameter_4,
```
You would just need to expand this PR to cover all 256 values. It's possible there may be some index values that go beyond 256, for example User Code CC v2. allows 65535 possible user code slots, will OZW have 64K possible indices?

Whatever the solution, also consider what happens when a client encounters undefined indices like this. For example, what happens when Home Assistant wants to add support for some particular index that isn't defined here? The process involves:

1. adding the enum definition here
2. releasing a new version here
3. updating HA to use the new version
4. then finally updating HA to use the new enum

It would be nice for clients to be functional w/o requiring the enum value definition.

Reply:
Maybe just change value.py L#76:
```
    @property
    def index(self) -> ValueIndex:
        """Return Index."""
        try:
            # TODO: we can make this prettier by returning the
            # ValueIndex belonging to this specific CC
            # Now it will return the first value matching this int value
            # Nothing breaks but it isn't very pretty
            return ValueIndex(self.data.get("Index"))
        except ValueError:
            return self.data.get("Index")
```
until there's a better way?